src/authConfig.ts

```diff
@@ -16,37 +19,79 @@ declare global {
     }
 }
 
-export const createMsalConfig = (clientId: string, authority: string, redirectUri: string): Configuration => {
-    return {
-        auth: {
-            // the following two values are changed using the Environment Provider in the layout.tsx
-            clientId,
-            authority, // This is a URL (e.g. https://login.microsoftonline.com/{your tenant ID})
-            redirectUri,
+const keycloakEnabled = process.env.KEYCLOAK_ENABLED?.toLowerCase() === 'true'.toLowerCase();
+
+export const authOptions: AuthOptions = {
+    providers: [
+        ...(keycloakEnabled ? [
+            KeycloakProvider({
+                clientId: process.env.KEYCLOAK_CLIENT_ID ? process.env.KEYCLOAK_CLIENT_ID : '',
+                clientSecret: '-', // not required by the AuthFlow but required by NextAuth Provider, here placeholder only
+                issuer: process.env.KEYCLOAK_ISSUER
+            })
+        ] : [
+            AzureADProvider({
+                clientId: process.env.AD_CLIENT_ID ? process.env.AD_CLIENT_ID : '',
+                clientSecret: process.env.AD_SECRET_VALUE ?? '',
+                tenantId: process.env.AD_TENANT_ID,
+                authorization: { params: { scope: `openid ${process.env.APPLICATION_ID_URI}admin.write` } },
+            })
+        ])
+    ],
+    session: {
+        strategy: 'jwt',
+    },
+    callbacks: {
+        async jwt({ token, account }) {
+            const nowTimeStamp = Math.floor(Date.now() / 1000);
+
+            if (account) {
+                token.access_token = account.access_token;
+                token.id_token = account.id_token;
+                token.expires_at = account.expires_at;
+                token.refresh_token = account.refresh_token;
+                return token;
+            } else if (nowTimeStamp < (token.expires_at as number)) {
+                return token;
+            } else {
+                if(!keycloakEnabled) return token;
+                try {
+                    console.warn('Refreshing access token...');
+                    return await refreshAccessToken(token);
+                } catch (error) {
+                    console.error('Error refreshing access token', error);
+                    return { ...token, error: 'RefreshAccessTokenError' };
+                }
+            }
```

Comment:
You could remove the else block here to reduce nesting
```suggestion
            }
            
            if (!keycloakEnabled) return token;
                            
            try {
                console.warn('Refreshing access token...');
                return await refreshAccessToken(token);
            } catch (error) {
                console.error('Error refreshing access token', error);
                return { ...token, error: 'RefreshAccessTokenError' };
            }
```